web/vibe/web/common.d

```diff
@@ -492,25 +492,70 @@ package struct MethodAttribute
 	alias data this;
 }
 
+
+/** Uses a custom serializer for the method return value.
+
+	Instead of using the default serializer (JSON), this allows to define
+	custom serializers. Multiple serializers can be specified and will be
+	matched against the `Accept` header of the HTTP request.
+*/
+alias resultSerializer(alias serialize, alias deserialize, string content_type)
+	= ResultSerializer!(serialize, deserialize, content_type);
+
 ///
-public struct ResultSerializer (alias ST, alias DT, string ContentType)
+unittest {
+	import std.bitmanip : bigEndianToNative, nativeToBigEndian;
+
+	interface MyRestInterface {
+		static struct Point {
+			int x, y;
+		}
+
+		static void serialize(R)(ref R output_range, ref Point value)
+		{
+			output_range.put(nativeToBigEndian(value.x));
+			output_range.put(nativeToBigEndian(value.y));
+		}
```

Comment:
Can you extend the example to use an overload set (on `value`) ?
Also, maybe make `value` be `const` ?

Reply:
Made it const and turned it into a template parameter. Overload set would also be possible with the latest changes, that could maybe be added as a second example? But it does get quite lengthy then.